modules/tensor_mechanics/src/materials/crystal_plasticity/CrystalPlasticityStressUpdateBase.C

```diff
@@ -119,14 +119,21 @@ CrystalPlasticityStressUpdateBase::CrystalPlasticityStressUpdateBase(
 void
 CrystalPlasticityStressUpdateBase::initQpStatefulProperties()
 {
-  _tau[_qp].resize(_number_slip_systems);
-
-  _flow_direction[_qp].resize(_number_slip_systems);
+  setMaterialVectorSize();
   for (const auto i : make_range(_number_slip_systems))
   {
     _flow_direction[_qp][i].zero();
     _tau[_qp][i] = 0.0;
   }
```

Comment:
resize and initialization should be done within the same function. Maybe consider moving the initialization to the new function too.

Reply:
@jiangwen84 
I have incorporated these changes in the new commit.